File: src/core/DiscardPile.java
package core;

/**
 * Pila de cartas descartadas de la partida.
 * Es donde los jugadores pondrán sus cartas a medida que avanza la partida.
 */
public class DiscardPile extends Deck{
	
	private Deck deck;
	/**
	 * Constructor de la clase.
	 * @param deck Baraja que se utilizará en la partida.
	 */
	public DiscardPile(Deck deck) {
		this.cards.clear();
		this.setDeck(deck);
		
	}
    /**
     * Instancia la DiscardPile con las cartas especificadas.
     * 
     * @param card Cartas requeridas en la DiscardPile.
     * Se debe validar con la expreción regula contenida en ERegex.DECK.
     */	
	public DiscardPile (String cards) {
		super(cards);
	}
	/**
	 * Recive una carta específica de la mano de un jugador.
	 * @param hand Mano del jugador que dará la carta.
	 * @param card Carta que el jugadoro dará.
	 */
	public void receiveCard(Hand hand, Card card) {
		this.cards.add(hand.giveCard(card));		
	}
	
	/**
	 * Recive una carta aleatoria no especial para iniciar la partida.
	 */
	public void receiveFirstCard() {
		boolean x = false;
		
		while (!x) {
			try {
				this.cards.add(deck.giveCard(RandomGenerator.noActionUNOCard()));	
				x = true;
			} catch (Exception e) {
				x = false;
			}			
		}
	}
	
	/**
	 * @return the deck
	 */
	public Deck getDeck() {
		return deck;
	}
	/**
	 * @param deck the deck to set
	 */
	public void setDeck(Deck deck) {
		this.deck = deck;
	}
	
	public static void main(String[] args) {
		Deck a = new Deck();
		DiscardPile ad = new DiscardPile(a);
		DiscardPile bd = ad;
		System.out.println(ad.equals(bd));
		
		bd = new DiscardPile(a.shuffle());
		System.out.println(ad.equals(bd));
		
		
	}
}

File: WebContent/core2/DiscardPile.js
/**
 * Pila de cartas descartadas de la paradida.
 * Es donde los jugadores pondrán sus cartas a medida que avanza el juego.
 * @param {Deck} deck Baraja asociada a la pila.
 */
function DiscardPile(deck=null,generated=false){
    Deck.call(this,generated);

    this.deck = deck;

	/**
	 * Recive una carta específica de la mano de un jugador.
	 * @param {Hand} hand Mano del jugador que dará la carta.
	 * @param {Card} card Carta que el jugadoro dará.
	 */
    this.receiveCard=function(hand,card){
        var cardIndex = hand.searchIndexCard(card);
        if (cardIndex == false){
            throw "La carta no se encuentra la mano.";
        }
        this.cards.add(hand.giveCardbyIndex(cardIndex));
    }

	/**
	 * Recive una carta aleatoria no especial para iniciar la partida.
     * @param {Deck} deck Baraja de donde se sacará la primera carta.
	 */
    this.receiveFirstCard = function(deck = this.deck){
        if (deck.cards.length() < 10){
            throw "Pocas cartas en la baraja.";
        }
        var r = new RandomGenerator();
        var card = deck.giveCard(r.noActionUNOCard());
        while(card == false){
            card = deck.giveCard(r.noActionUNOCard());
        }

        this.cards.add(card);
    }
}